exporters/otlp/README.md

```diff
@@ -3,13 +3,146 @@
+		}
+	}()
+
+	// Note: The exporter can also be used as a Batcher.
+	traceProvider, err := sdktrace.NewProvider(sdktrace.WithSyncer(exporter))
+	if err != nil {
+		log.Fatal("failed to create trace provider: %v", err)
+	}
+
+	pusher := push.New(simple.NewWithExactDistribution(), exporter)
+	pusher.Start()
+	metricProvider := pusher.Provider()
+
+	// Your code here ...
+}
+```
+
+## Configuration
+
+Configurations options can be specified when creating a new exporter (`NewExporter`).
+
+### `WorkerCount(n uint)`
+
+Sets the number of Goroutines to use when processing telemetry.
+
+
+### `WithInsecure()`
+
+Disables client transport security for the exporter's gRPC connection just like [`grpc.WithInsecure()`](https://pkg.go.dev/google.golang.org/grpc#WithInsecure) does.
+By default, client security is required unless `WithInsecure` is used.
+
+### `WithAddress(addr string)`
+
+Sets the address that the exporter will connect to the collector on.
+The default address the exporter connects to is `localhost:55680`.
+
+### `WithReconnectionPeriod(rp time.Duration)`
+
+Set the delay between connection attempts after failing to connect with the collector.
+
+### `WithCompressor(compressor string)`
+
+Set the compressor for the gRPC client to use when sending requests.
+The compressor used needs to have been registered with `google.golang.org/grpc/encoding` prior to using here.
+This can be done by `encoding.RegisterCompressor`.
+Some compressors auto-register on import, such as gzip, which can be registered by calling `import _ "google.golang.org/grpc/encoding/gzip"`.
+
+### `WithHeaders(headers map[string]string)`
+
+Headers to send when the gRPC stream connection is instantiated.
+
+### `WithTLSCredentials(creds "google.golang.org/grpc/credentials".TransportCredentials)`
+
+TLS credentials to use when talking to the server.
+
+### `WithGRPCServiceConfig(serviceConfig string)`
+
+The default gRPC service config used when .
+
+By default, the exporter is configured to support [retries](#retries).
+
+```json
+{
+	"methodConfig":[{
+		"name":[
+			{ "service":"opentelemetry.proto.collector.metrics.v1.MetricsService" },
+			{ "service":"opentelemetry.proto.collector.trace.v1.TraceService" }
+		],
+		"waitForReady": true,
+		"retryPolicy":{
+			"MaxAttempts":5,
+			"InitialBackoff":"0.3s",
+			"MaxBackoff":"5s",
+			"BackoffMultiplier":2,
+			"RetryableStatusCodes":[
+				"UNAVAILABLE",
+				"CANCELLED",
+				"DEADLINE_EXCEEDED",
+				"RESOURCE_EXHAUSTED",
+				"ABORTED",
+				"OUT_OF_RANGE",
+				"UNAVAILABLE",
+				"DATA_LOSS"
+			]
+		}
+	}]
+}
+```
+
+### `WithGRPCDialOption(opts ..."google.golang.org/grpc".DialOption)`
+
+Additional `grpc.DialOption` to be used.
+
+These options take precedence over any other set by other parts of the configuration.
+
+## Retries
+
+The exporter will not, by default, retry failed requests to the collector.
+However, it is configured in a way that it can easily be enable.
```

Comment:
That's unfortunate. I guess `os.Setenv("GRPC_GO_RETRY", "on")` would work.

Reply:
FYI there is also a retry middleware that we can use by default or enable via config flag https://github.com/grpc-ecosystem/go-grpc-middleware/tree/master/retry

The `GRPC_GO_RETRY` environment variable is global and will affect all gRPC clients used by the application, not just OTLP exporter

Reply:
> That's unfortunate. I guess `os.Setenv("GRPC_GO_RETRY", "on")` would work.

I tried this as well. The problem though is the Go runtime loading order. The dependencies (and consequentially the evaluation of gRPCs environment variables) are evaluated prior to being able to set the environment variable.

> FYI there is also a retry middleware that we can use by default or enable via config flag https://github.com/grpc-ecosystem/go-grpc-middleware/tree/master/retry

I looked into this as an option as well. I didn't want to add an additional dependency, and I wanted to support what was being asked in the related ticket, that all connections to the collector be configured to support the retry mechanism.

> The GRPC_GO_RETRY environment variable is global and will affect all gRPC clients used by the application, not just OTLP exporter

That is correct, however, the service config uses service name pattern matching to narrow down the scope. In this case it will be for all connections to the collector.

Reply:
Sorry, I failed to make it clear in my last comment, but I am open to using https://github.com/grpc-ecosystem/go-grpc-middleware/tree/master/retry if there is a strong opinion that it is the preferred solution here.

Reply:
Using retry middleware makes sense if you want to have retry mechanism enabled by default without requiring users to define an environment variable.

Otherwise should be fine to have the GRPC_GO_RETRY env var documented.

Reply:
@nilebox I think keeping the current solution as an initial solution is a way to move this issue forward and progressing the milestone. I've created https://github.com/open-telemetry/opentelemetry-go/issues/789 to track switching to the middleware in the future.